### PR TITLE
takeaway challenge

### PR DESCRIPTION
**Pull request by HuxleyM**
i made two versions, v2 with 3 classes  i ran into difficulty using rspec but much works bar texting and the other version is simpler two classes. Neither finished :/

### CODE REVIEW

File: lib/menu.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        @list = [
+            {:dish => 'green curry', :price => 5.99},
+            {:dish => 'panang', :price => 7.5},
+            {:dish => 'veg tempura', :price => 4.5},
+            {:dish => 'rice', :price => 2.5}
+            ]
+        @choices = []
+    end
+
+    def select(dish_string, quantity)
```
Comment (stonefarmer9) [inline lib/menu.rb]: Is this th responsibility of the menu class or the order class?

File: lib/menu.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        @list.each do |details|
+            if details[:dish] == dish_string
+                details.merge!({:quantity => quantity})
+                return @choices.push(details)
```
Comment (stonefarmer9) [inline lib/menu.rb]: This is come nice code and I think better than the nested each methods I used!

File: lib/order.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Order 
+    attr_reader :dishes, :total
+
+    def initialize(menuChoice_obj)
```
Comment (stonefarmer9) [inline lib/order.rb]: Is menu choice_obj a clear name? I

File: spec/menu_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+describe Menu do 
+  let(:thai_menu) {Menu.new}
+  let(:green_curry_details) {{:dish => 'green curry', :price => 5.99}}
+  let(:dish) {'green curry'}
+  let(:quantity) {2}
+  
+  context ' initialize' do
+
+    it 'on #display will show list of menu items' do 
+        expect(thai_menu.list).to include(green_curry_details)
```
Comment (stonefarmer9) [inline spec/menu_spec.rb]: Is testing an expected inclusion as rigourous as testing a full equality exepctatiom?

File: spec/menu_spec.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    it 'should have a #select method that will ask for two params' do 
+      expect(thai_menu).to respond_to(:select).with(2)
+    end
+
+    it' should return the array for the specified dish from @list' do 
+        expect(thai_menu.select(dish, quantity )).to eql([green_curry_details_and_quantity])
+    end
+
+    it 'the returned array should include a quanitity key pair made from user input' do 
+    end
```
Comment (stonefarmer9) [inline spec/menu_spec.rb]: What does this test?

File: spec/order_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    end
+
+    describe '#place_order' do 
+      it 'expect a param of a customers phone number' do 
+         expect(order).to respond_to(:place_order).with(1)
+      end
+    end 
+
+
+end
```
Comment (subomionanuga) [inline spec/order_spec.rb]: Consider adding a README

File: lib/menu.rb (hunk 1 above)
Comment (CelineKaslin) [inline lib/menu.rb]: Not sure that the class menu has the responsibility of select the dishes. 
Maybe an other class like order or basket would have been a way to approach the challenge

File: lib/order.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        @dishes = menuChoice_obj
+        @total = 0
+    end
+
+    def sum
```
Comment (CelineKaslin) [inline lib/order.rb]: maybe a separate class for sum will have been an idea to separate the responsibility, and the class order will maybe have encapsulate the menu, eventually basket for the selection of the dishes and the sum class for the total.

File: lib/order.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+       @dishes.each do |item|
+           @total += item[:price]
+       end
+       return @total
+    end
+
+    def place_order(phone_string)
```
Comment (CelineKaslin) [inline lib/order.rb]: sorry I might have miss something, but what is the phone_string parameter ?

File: send_sms.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Download the twilio-ruby library from twilio.com/docs/libraries/ruby
+require 'twilio-ruby'
+
```
Comment (CelineKaslin) [inline send_sms.rb]: are those method wrapped in a class ?

File: spec/menu_spec.rb (hunk 5 above)
Comment (CelineKaslin) [inline spec/menu_spec.rb]: Same question as Stonefarmer9, what is the purpose of this test ? What do you expect from the subject ?

File: README.md [hunk 10]
```diff
@@ -13,6 +13,45 @@ Takeaway Challenge
...
+
+Menu includes :Select allowing us to choose a dish
+ 
+Order includes the functionality to :add, :sum and :close.
+
+Customer includes : check_menu, :add, check_order_total, :new_order : place_order
+
+all methods are designed to be single responsiblity. I made Order a part of customer as they couldnt exist without each other. 
+
+is the logic ok??
```
Comment (rednblack99) [inline README.md]: I really like this approach but I can see why it was tricky to test. My only suggestion for a solution is to double every single one of the hashes to predict what they would contain but that would be very tricky to write.

File: lib/menu.rb [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+            end
+        end
+        raise 'dish not found'
+    end
+end
```
Comment (rednblack99) [inline lib/menu.rb]: Did you run Rubocop? There are a few minor formatting issues such as not having the newlines at the end of your files to ensure it's compliant with the usual Ruby approach.

**Pull request closed without merging**